test-lib-openshift.sh

```diff
@@ -694,3 +756,156 @@ ct_os_test_image_update() {
 
   ct_os_delete_project
 }
+
+# ct_os_deploy_cmd_image IMAGE_NAME
+# --------------------
+# Runs a special command pod, a pod that does nothing, but includes utilities for testing.
+# A typical usage is a mysql pod that includes mysql commandline, that we need for testing.
+# Running commands inside this command pod is done via ct_os_cmd_image_run function.
+# The pod is not run again if already running.
+# Arguments: image_name - image to be used as a command pod
+function ct_os_deploy_cmd_image() {
+  local image_name=${1}
+  oc get pod command-app &>/dev/null && echo "command POD already running" && return 0
+  echo "command POD not running yet, will start one called command-app"
+  oc create -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: command-app
+spec:
+  containers:
+  - name: command-container
+    image: ${image_name}
+    command: ["sleep"]
+    args: ["3h"]
+  restartPolicy: OnFailure
+EOF
+
+  SECONDS=0
+  echo -n "Waiting for command POD ."
+  while [ $SECONDS -lt 60 ] ; do
+    ct_os_cmd_image_run "echo 24" &>/dev/null && echo "DONE" && return 0
+    sleep 3
+    echo -n "."
+  done
+  echo "FAIL"
+  return 1
+}
+
+# ct_os_cmd_image_run CMD [ ARG ... ]
+# --------------------
+# Runs a command CMD inside a special command pod
+# Arguments: cmd - shell command with args to run in a pod
+function ct_os_cmd_image_run() {
```

Comment:
What about automatic calling `ct_os_deploy_cmd_image` on first `ct_os_cmd_image_run` invocation?

Reply:
`ct_os_deploy_cmd_image` requires an image name to be used, so we would either need to specify image for each `ct_os_cmd_image_run` call, or run a default (python) image for those cases. I don't have a strong opinion about which variant is better, or whether the current solution is enough.. Feel free to express your preference.

Reply:
You are right. Maybe better to require explicit call to deploy image.